etna/analysis/forecast/plots.py

```diff
@@ -35,6 +37,26 @@
     from etna.transforms import Transform
 
 
+def _get_borders_comparator(segment_borders: pd.DataFrame) -> Callable[[str, str], int]:
+    """Create comparator function to sort border names."""
+
+    def cmp(name_a: str, name_b: str) -> int:
+        """Compare two series based on their values."""
+        border_a = segment_borders[name_a].values
+        border_b = segment_borders[name_b].values
+
+        if np.all(border_a < border_b):
```

Comment:
What if `np.all(border_a <= border_b)`, but therer is one index for which `border_a < border_b`?